### PR TITLE
Upgrade to Blacklight 6 and Rails 5

### LINKED ISSUE #126: Upgrade to BL Version 6
Current deprecation warning under 5.18

```
DEPRECATION WARNING: Your application is missing the SearchBuilder. Have you run `rails generate blacklight:search_builder`? Falling back to Blacklight::Solr::SearchBuilder. (called from locate_search_builder_class at /opt/lae/shared/bundle/ruby/2.1.0/gems/blacklight-5.18.0/lib/blacklight/configuration.rb:228)
```

### PR DESCRIPTION
**Pull request by eliotjordan**
Closes #126

### CODE REVIEW

File: app/views/_user_util_links.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div>
   <ul class="nav navbar-nav">
-    <% if render_bookmarks_control? %>
+    <% if has_user_authentication_provider? && current_or_guest_user.present? %>
```
Comment (escowles) [inline app/views/_user_util_links.html.erb]: Is this the logic that used to be in `render_bookmarks_control?` ?

File: solr/conf/schema.xml [hunk 2]
```diff
@@ -177,11 +160,9 @@
       <analyzer>
         <!-- <tokenizer class="solr.WhitespaceTokenizerFactory"/> -->
         <tokenizer class="solr.StandardTokenizerFactory"/>
-        <filter class="solr.ICUFoldingFilterFactory" />
```
Comment (escowles) [inline solr/conf/schema.xml]: Don't we need this to handle Spanish correctly?

**Comment by escowles** [comment]
re: the weird broken RDF/XML — I guess the first thing is to make sure we need RDF/XML support.  The Turtle and JSON-LD are fine, and in Plum we've been doing those two plus N-Triples, and haven't been doing RDF/XML.

File: app/views/_user_util_links.html.erb (hunk 1 above)
Comment (eliotjordan) [inline app/views/_user_util_links.html.erb]: Yes. `render_bookmarks_control?` was moved into the controller.
https://github.com/projectblacklight/blacklight/blob/master/app/controllers/concerns/blacklight/controller.rb#L64

File: solr/conf/schema.xml (hunk 2 above)
Comment (eliotjordan) [inline solr/conf/schema.xml]: Good catch. I was just hacking through the schema to get it working with solr_wrapper/Solr 6. I'm not sure how to implement this to satisfy old and new solr versions. I'll look into it.

**Comment by eliotjordan** [comment]
@jpstroop @kevinreiss Do we need RDF/XML support in lae-blacklight (see above)? Upgrading the rdf gems for Rails 5 causes an issue with this, and it would be good to know if it's worth spending the time tracking down the error.

File: solr/conf/schema.xml (hunk 2 above)
Comment (eliotjordan) [inline solr/conf/schema.xml]: This might be the issue: https://issues.apache.org/jira/browse/SOLR-6188
I'll try using the fully qualified class name.

**Comment by eliotjordan** [comment]
Updated solrconfig to match the configs in `pul_solr`. There are a few changes in the config file that are incompatible with earlier versions of Solr. The app should continue to work with the current lae Solr instances, but I created a follow-on issue to migrate to lib-solr1 and lib-solr2:

https://github.com/pulibrary/lae-blacklight/issues/145

**Comment by escowles** [comment]
I just pushed a commit to switch to the latest development branch of `rdf-rdfxml` — looks like there's a HAML bug causing the invalid RDF/XML output that it's working around: https://github.com/ruby-rdf/rdf-rdfxml/commit/9f43ed940d0122f1c9b43f255ad4b558455708ea

**Pull request merged**